src/icons/general/IconPuzzlePieceCircle.js

```diff
@@ -0,0 +1,28 @@
+// @flow
+import * as React from 'react';
+import AccessibleSVG from '../accessible-svg';
+import { bdlBoxBlue } from '../../styles/variables';
+
+type Props = {
+    className?: string,
+    color?: string,
+    dimension?: number,
+    title?: string | React.Element<any>,
+};
+
+const IconPuzzlePieceCircle = ({ className = '', color = bdlBoxBlue, dimension = 28, title }: Props) => (
+    <AccessibleSVG
+        className={`bdl-IconPuzzlePieceCircle ${className}`}
```

Comment:
minor: you could use the `classname` library to make this `classNames('bdl-IconPuzzlePieceCircle', className)` which helps avoid an extra space in the rendered class if no `className` is provided